app/Http/Controllers/Core/ProductController.php

```diff
@@ -57,18 +58,34 @@ public function productShowNew()
         return view('product_new', $data);
     }
 
-    public function productShowEdit($product_name)
+    public function productShowEdit($product_code)
     {
-        $data['product'] = $this->productService->getOne($product_name);
+        $data['product'] = $this->productService->getOne($product_code);
         $data['providers'] = $this->providerService->getAll();
         return view('product_edit', $data);
     }
 
-    public function productEdit($product_name, Request $request)
+    public function productEdit($product_code, Request $request)
     {
-        $product = $this->productService->getOne($product_name);
+        $product = $this->productService->getOne($product_code);
 
         $this->productService->productEdit($product, $request);
         return redirect('admin/product');
     }
+
+    public function getProductsByPhoneNumber(Request $request)
+    {
+        $providers = Provider::all();
+        $phonePrefix = substr($request->get('nomor_hp'), 0, 4);
+        $productArray = [];
+        foreach ($providers as $provider) {
+             $arrays = explode(',', $provider->prefixes);
+             for ($i = 0; $i < count($arrays); $i++) {
```

Comment:
1. `arrays` nya diganti yang lebih self explanatory, mungkin `providerPrefixes`
2. ini buat ngecek apakah prefix dari input user sama dengan list prefix dari provider yang sedang di foreach kan ?
bisa make [in_array($needle, $haystack)](http://php.net/manual/en/function.in-array.php)
```php
if (in_array($phonePrefix, $providerPrefixes)) {
  $productArray  = $provider->products;
  break; // ini biar keluar dari foreach, kalo udah ketemu, ga perlu cek provider selanjutnya
}
```
3. ini belum ngebedain antara pulsa dan data, view nya juga harus ditambahin